tests/integration/test_models/standard_output_comparison.py

```diff
@@ -0,0 +1,181 @@
+        "Run all tests from a class 'ClassName'"
+        if skip_first_timestep:
+            t = self.t[1:]
+        else:
+            t = self.t
+        tests = ClassName(self.models, t, self.mesh, self.solvers)
+        tests.test_all()
+
+    def test_averages(self, skip_first_timestep=False):
+        self.run_test_class(AveragesComparison, skip_first_timestep)
+
+    def test_all(self, skip_first_timestep=False):
+        self.test_averages(skip_first_timestep)
+        self.run_test_class(VariablesComparison, skip_first_timestep)
+
+        if self.chemistry == "Lithium-ion":
+            self.run_test_class(ParticleConcentrationComparison)
+        elif self.chemistry == "Lead-acid":
+            self.run_test_class(PorosityComparison)
+
+
+class BaseOutputComparison(object):
+    def __init__(self, models, time, mesh, solvers):
+        self.models = models
+        self.t = time
+        self.mesh = mesh
+        self.solvers = solvers
+
+    def get_vars(self, var):
+        "Helper function to reduce repeated code."
+        return [
+            pybamm.ProcessedVariable(
+                model.variables[var],
+                self.solvers[model].t,
+                self.solvers[model].y,
+                mesh=self.mesh,
+            )
+            for model in self.models
+        ]
+
+    def compare(self, var, tol=1e-2):
+        "Compare variables from different models"
+        # Get variable for each model
+        model_variables = self.get_vars(var)
+        var0 = model_variables[0]
+
+        if var0.domain == []:
+            x = None
+        else:
+            x = self.mesh.combine_submeshes(*var0.domain)[0].nodes
+
+        # Calculate tolerance based on the value of var0
+        maxvar0 = np.max(abs(var0(self.t, x)))
+        if maxvar0 == 0:
+            decimal = -int(np.log10(tol))
+        else:
+            decimal = -int(np.log10(tol * maxvar0))
+        # Check outputs are close to each other
+        for model_var in model_variables[1:]:
+            np.testing.assert_equal(var0.dimensions, model_var.dimensions)
+            np.testing.assert_array_almost_equal(
+                model_var(self.t, x), var0(self.t, x), decimal
+            )
+
+
+class AveragesComparison(BaseOutputComparison):
+    "Compare variables whose average value should be the same across all models"
+
+    def __init__(self, models, time, mesh, solvers):
+        super().__init__(models, time, mesh, solvers)
+
+    def test_all(self):
+        # Potentials
+        self.compare("Average open circuit voltage")
+        # Currents
+        self.compare("Average negative electrode interfacial current density")
+        self.compare("Average positive electrode interfacial current density")
+        # Concentration
+        self.compare("Average electrolyte concentration")
+
+
+class VariablesComparison(BaseOutputComparison):
+    "Compare variables that vary across models"
```

Comment:
I don't get this.... doesn't `self.compare` test that the variables are similar across models? I feel I'm missing something obvious

Reply:
Yeah re-reading now this is bad wording ... the distinction between `VariablesComparison` and `AveragesComparison` is that averages should be the same across models regardless of current, whereas variables will differ a bit and only be approximately equal for certain (low) values of the current. 
May this should just be "compare variables across models"?

Reply:
Yeah re-reading now this is bad wording ... the distinction between VariablesComparison and AveragesComparison is that averages should be the same across models regardless of current, whereas variables will differ a bit and only be approximately equal for certain (low) values of the current.
May this should just be "compare variables across models"?

Reply:
@martinjrobins - I have changed to "compare variables across models", let me know if this is ok and I will merge